### PR TITLE
feat(manager/pip-compile): Allow security updates for transitive dependencies 

### PR DESCRIPTION
**Pull request by not7cd**
## Changes

Indirect dependencies that exist only in lock files are now added to the package file's dependency list. They are disabled by default, but this can be overwritten by the configuration, like vulnerability alerts.

## Context

closes https://github.com/renovatebot/renovate/pull/26871
closes https://github.com/renovatebot/renovate/discussions/26947

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/pip-compile/extract.ts [hunk 1]
```diff
@@ -206,3 +214,31 @@ export async function extractAllPackageFiles(
...
+}
+function indirectDep(dep: PackageDependency): PackageDependency {
+  const result = {
+    ...dep,
+    lockedVersion: dep.currentVersion,
+    depType: 'indirect',
+    enabled: false,
+  };
+  delete result.currentValue;
+  delete result.currentVersion;
```
Comment (viceice) [inline lib/modules/manager/pip-compile/extract.ts:242]: why this? 🤔

Comment (not7cd) [reply lib/modules/manager/pip-compile/extract.ts:242]: I'm removing these values, so this treated just like unconstrained dependency. There are some parts that give such dependencies "isLockfileUpdate" flag. So this ensures that. Maybe this object can be recreated in another way?

Comment (viceice) [reply lib/modules/manager/pip-compile/extract.ts:242]: are they set from extraction?

**Review by viceice — COMMENTED** [review]
see comment

File: lib/modules/manager/pip-compile/extract.ts (hunk 1 above)
Comment (not7cd) [reply lib/modules/manager/pip-compile/extract.ts:242]: Yes, they are parsed by `pip_requirements` manager. I want to keep everything else, so future metadata is not omitted. And just removing these two values makes it behave like any other unconstrained dependency when enabled.

Comment (not7cd) [reply lib/modules/manager/pip-compile/extract.ts:242]: Should I add a comment in code why these properties are removed?

**Comment by not7cd** [comment]
I would like to add that this change has been checked against over 100 repositories over past month. And I don't see any unwanted updates.

File: lib/modules/manager/pip-compile/extract.ts (hunk 1 above)
Comment (viceice) [reply lib/modules/manager/pip-compile/extract.ts:242]: yes please do that

**Review by viceice — COMMENTED** [review]
review is no longer applied by adding comment on existing discussion 😕

**Review by viceice — APPROVED** [review]

**Pull request merged**